### PR TITLE
[camera_platform_interface] Migrate to null safety

### PR DESCRIPTION
**Pull request by mvanbeusekom**
This PR is a follow up on PR #3497 and contains fixes for the following feedback:

- Improve documentation of `availableCameras` method ([original comment](https://github.com/flutter/plugins/pull/3497#discussion_r572389747));
- Removed dependency on Mockito ([original comment](https://github.com/flutter/plugins/pull/3497#discussion_r572394822));
- Use a default value of `false` for the `enableAudio` parameter ([original comment](https://github.com/flutter/plugins/pull/3497#discussion_r572396714));
- Add left hand type definition to declared variable ([original comment](https://github.com/flutter/plugins/pull/3497#discussion_r572424013)).

Note: the version information in the pubspec.yaml and CHANGELOG.md is not updated on purpose since this should be published with the same release as the original PR.

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [-] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [-] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/camera/camera_platform_interface/lib/src/method_channel/method_channel_camera.dart [hunk 1]
```diff
@@ -59,7 +59,7 @@ class MethodChannelCamera extends CameraPlatform {
   @override
   Future<List<CameraDescription>> availableCameras() async {
     try {
-      final cameras = await _channel
+      final List<Map<dynamic, dynamic>>? cameras = await _channel
```
Comment (ditman) [inline packages/camera/camera_platform_interface/lib/src/method_channel/method_channel_camera.dart:62]: This happened in several places across the code. I think as @bparrishMines suggested we should enable a lint to enforce either always adding (or never adding) this type.

**Review by ditman — APPROVED** [review]
Thanks for all the changes Maurits! And sorry for publishing late the comments to your earlier PR, yesterday I got late to *every* PR :/

**Comment by ditman** [comment]
```
Checking project camera_platform_interface...

  info - Unused import: 'package:plugin_platform_interface/plugin_platform_interface.dart' at test/camera_platform_interface_test.dart:9:8 - (unused_import)
```

File: packages/camera/camera_platform_interface/lib/src/method_channel/method_channel_camera.dart (hunk 1 above)
Comment (mvanbeusekom) [reply packages/camera/camera_platform_interface/lib/src/method_channel/method_channel_camera.dart:62]: Understood, I will leave it like this for now and when we decide on which version we will go for I will make sure everything is corrected.

**Comment by mvanbeusekom** [comment]
> ```
> Checking project camera_platform_interface...
> 
>   info - Unused import: 'package:plugin_platform_interface/plugin_platform_interface.dart' at test/camera_platform_interface_test.dart:9:8 - (unused_import)
> ```

Thanks, forgot to run analyze locally. Fixed it and waiting for tree to go green ;)

**Comment by mvanbeusekom** [comment]
Thank you very much for the review @ditman, better late then sorry ;) Really appreciate it.

**Comment by ditman** [comment]
I'll merge this as soon as the tree goes green again. Thanks for updating this to the latest master!

**Review by bparrishMines — APPROVED** [review]
LGTM

**Comment by mvanbeusekom** [comment]
> I'll merge this as soon as the tree goes green again. Thanks for updating this to the latest master!

Done and done ;-)

**Pull request merged**